### PR TITLE
Minor changes for Enzyme

### PR DESCRIPTION
**Pull request by swilliamson7**
These are a few updates that Enzyme needs in order to run. Specifically regarding the `mutable struct` changes, I'm working on understanding why this is needed but without it Enzyme crashes. Let me know if you have questions!

### CODE REVIEW

**Comment by swilliamson7** [comment]
Adding that after this is merged, could a new version of ShallowWaters be released? So that I can test some stuff for a presentation I need to give. I'm guessing that, if it doesn't work like I think it should we could always roll back to the last version

File: src/default_parameters.jl [hunk 1]
```diff
@@ -8,8 +8,8 @@
 
     # DOMAIN RESOLUTION AND RATIO
     nx::Int=100                         # number of grid cells in x-direction
-    Lx::Real=4000e3                     # length of the domain in x-direction [m]
-    L_ratio::Real=2                     # Domain aspect ratio of Lx/Ly
+    Lx::Int=4000e3                      # length of the domain in x-direction [m]
+    L_ratio::Int=2                      # Domain aspect ratio of Lx/Ly
```
Comment (milankl) [inline src/default_parameters.jl:12]: what about `Float64`? While for `Lx` integer is probably fine, `L_ratio = 1.5` does make sense to me and should be allowed?

File: src/default_parameters.jl [hunk 2]
```diff
@@ -20,7 +20,7 @@
     R::Real=6.371e6                     # Earth's radius [m]
 
     # SCALE
-    scale::Real=2^6                     # multiplicative scale for the momentum equations u,v
+    scale::Int=2^6                      # multiplicative scale for the momentum equations u,v
```
Comment (milankl) [inline src/default_parameters.jl:23]: I see that ideally you'd want scaling to be a power of two which is always an int, but it still works fine with `Float64` so let's not be unnecessarily restrictive?

**Comment by milankl** [comment]
> Adding that after this is merged, could a new version of ShallowWaters be released? So that I can test some stuff for a presentation I need to give. I'm guessing that, if it doesn't work like I think it should we could always roll back to the last version

Of course no problem. Happy with `mutable` but ideally fields should have concrete types, I know with `::Real` I didn't follow this rule but `Int` sounds too restrictive (because it could be a float and it would make sense and work) so should we not just say `Float64` for these?

**Comment by swilliamson7** [comment]
For both of the comments on variable type `Int`, this was originally just a quick fix to make some fields type stable, `Float64` should be just as okay!

How do I release a new version (assuming all the checks pass)?

**Comment by milankl** [comment]
While we're at it I just removed all abstract fields in `Parameter` and `Grid`.

**Comment by milankl** [comment]
A new release can be triggered on any commit (not pull request) with `@JuliaRegistrator register` however, we'll also need to first set the new version. The changes here are not breaking but in #175 did we cause a breaking change?

**Comment by swilliamson7** [comment]
> The changes here are not breaking but in https://github.com/milankl/ShallowWaters.jl/pull/175 did we cause a breaking change?

I can't find a good definition of a "breaking change" after looking through [this](https://juliareach.github.io/JuliaReachDevDocs/latest/release/) and a couple other sites, but I _think_ this means that there might be compatibility issues or something, like the user would need to update to a different Julia version to use ShallowWaters.jl maybe? 

In any case, the single struct change should not have caused this sort of change, later changes to include Enzyme might

**Comment by milankl** [comment]
A breaking change is a new version that is not backwards compatible with previous versions. For example if we change `run_model(a,b,c; g=10)` to `run_model(a,b,c; gravity=10)` that's breaking because every code that was written with the previous version using `g` would throw an error if the keyword `g` doesn't exist anymore (because it's now `gravity`). If you implement similar changes within the code that is not supposed to be used by the user then this is fine. There's also a grey zone depending on the outputs. If `f(x)` returns `a` in a previous version but then it returns `a, b` it's a breaking change as not something of the type of `a` is returned but a tuple meaning that any function using `a` wouldn't work as before. However, changing `g=10` to `g=9.81` I wouldn't consider a breaking change because while the simulation is entirely different it's qualitatively similar, almost statistically similar and definitely returns the same types and sizes of these types.

**Comment by swilliamson7** [comment]
Got it, that makes sense! Then to answer your original question the single struct modification is definitely not a breaking change

**Comment by milankl** [comment]
Yes, that's why I have increased the version to v0.5.2 not v0.6 ;)

**Comment by milankl** [comment]
I've just triggered the release on the commit that merged this pull request into main https://github.com/milankl/ShallowWaters.jl/commit/7d4296aa5b71c2a2f8293f2c11f5c7bbad2dc836#commitcomment-141817250

**Comment by swilliamson7** [comment]
Amazing, thank you! To trigger this was the only thing that needed to change the version line in Project.toml?

**Comment by milankl** [comment]
No you have to comment on a commit with `@ JuliaRegistrator register` (remove space between @ and J, adding here to avoid it commenting here again). It's a github bot that checks when it's mentioned somewhere and then pulls the current version over into the Julia general registry!

**Pull request merged**